### PR TITLE
Systemtest

### PR DESCRIPTION
**Pull request by walshmm**
Merging system tests back into `next` now that jobs are passing so we may resume normal development habits by working out of `next`.

Contains various changes to how we build the code as well as a battery of tests to be ran externally against running docker containers.

### CODE REVIEW

**Review by wdzhou — COMMENTED** [review]
Nice job!

Most of my review comments are questions because I am not familiar with these techniques as you.

One more question:  why do you still keep some tests that are empty?  For example
https://github.com/neutrons/data_workflow/blob/systemtest/tests/data/TEST/shared/autoreduce/reduce_TEST.py

File: Dockerfile.autoreducer [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+# create startup script
+RUN echo "#!/bin/bash" > /usr/bin/run_postprocessing && \
+    echo "/opt/postprocessing/queueProcessor.py &" >> /usr/bin/run_postprocessing && \
+    echo "sleep 1" >> /usr/bin/run_postprocessing && \
+    echo "tail -F /opt/postprocessing/log/postprocessing.log" >> /usr/bin/run_postprocessing && \
+    chmod +x /usr/bin/run_postprocessing
+
+# start the service
+CMD run_postprocessing
```
[the remark this answers is not in the document]
Comment (walshmm) [reply Dockerfile.autoreducer:42]: I think @peterfpeterson might have an answer for that.

File: .github/workflows/systemtests.yml [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+    - name: Build test data for autoreducer
+      run: make SNSdata.tar.gz
+    - name: Stand up docker containers
+      run: LDAP_SERVER_URI=. LDAP_DOMAIN_COMPONENT=. docker-compose up --build -d
+      env:
+          DJANGO_SETTINGS_MODULE: reporting.reporting_app.settings.unittest
+    - name: Wait for everything to start up
+      run: sleep 20
+    - name: Test with pytest
+      run: python -m pytest tests
```
[the remark this answers is not in the document]
Comment (walshmm) [reply .github/workflows/systemtests.yml:31]: I think so, looking at the files in the directory, the only ones that contain actual python tests are the system tests.  The individual sub modules seem to house their own unit tests

File: Dockerfile.webmon [hunk 3]
```diff
@@ -8,9 +8,17 @@ WORKDIR /usr/src/webmon
 # copy the necessary wheels and the Makefile which knows the dependency order
 COPY ./src/webmon_app/dist/django_nscd_webmon-*-none-any.whl .
 COPY ./src/workflow_app/dist/django_nscd_workflow-*-none-any.whl .
-COPY ./src/Makefile .
+COPY ./Makefile .
 
 # move the entry-point into the volume
 COPY ./src/webmon_app/docker-entrypoint.sh /usr/bin/docker-entrypoint.sh
+
+# RUN useradd -ms /bin/bash user
```
[the remark this answers is not in the document]
Comment (walshmm) [reply Dockerfile.webmon:16]: No, I will clean them up

**Comment by walshmm** [comment]
> Nice job!
> 
> Most of my review comments are questions because I am not familiar with these techniques as you.
> 
> One more question: why do you still keep some tests that are empty? For example https://github.com/neutrons/data_workflow/blob/systemtest/tests/data/TEST/shared/autoreduce/reduce_TEST.py

As for this, I think theses files are generated as part of a test for the reduction page.(https://monitor.sns.gov/reduction/test/)

According to the text on that page, 
```
- A default script must be placed in /SNS/TEST/shared/autoreduce/reduce_TEST_default.py
- A script template must be placed in /SNS/TEST/shared/autoreduce/reduce_TEST.py.template
- Review the parameters below and modify as needed. Hit enter to apply your changes before clicking the submit button.
- The reduce_TEST.py will automatically be overwritten once you click the submit button.
- Make sure all your template properties appear below: if not, contact 
```

so this is likely following this pattern.
